src/official-account/official-account.ts

```diff
@@ -252,6 +256,59 @@ class OfficialAccount extends EventEmitter {
     return ret.body
   }
 
+  async uploadTempFile (args: {
+    file: FileBox,
+    touser: string,
+    msgtype: OAMessageType
+  }): Promise<ErrorPayload> {
+    log.verbose('OfficialAccount', 'uploadTempFile(%s)', JSON.stringify(args))
+    const msgSet: Set<string> = new Set(['image', 'video', 'voice'])
+    if (!msgSet.has(args.msgtype)) {
+      throw Error(`msgtype <${args.msgtype}> is  not supported`)
+    }
+
+    // request header is not application/json
+    const simpleUnirest = getSimpleUnirest(WX_SERVER_ADDRESS, { 'Content-Type': 'multipart/form-data' })
+
+    // check and create cache dir to save temp file
+    const wechatyDirExist = fs.existsSync('.wechaty')
+    if (!wechatyDirExist) {
+      fs.mkdirSync('.wechaty')
+    }
+    const wechatyCacheDirExist = fs.existsSync('.wechaty/cache')
+    if (!wechatyCacheDirExist) {
+      fs.mkdirSync('.wechaty/cache')
+    }
+
+    const cacheFile = `.wechaty/cache/${UUID.v1()}-${args.file.name}`
+    await args.file.toFile(cacheFile, true)
+
+    const mediaResponse = await simpleUnirest.post(`media/upload?access_token=${this.accessToken}&type=${args.msgtype}`).attach('remote file', fs.createReadStream(cacheFile))
```

Comment:
the result of fileBox.toStream() is not supported at unirest. So, I make it a cache file to unify the file sending mechanism.

Reply:
Could we use a `Stream` instead of a `File` for this feature, which will be a better way to do this?

See: 

- https://github.com/wechaty/wechaty-freshdesk/blob/25502a201b41560f4983dfe4b536a674756963a7/src/freshdesk/api.ts#L86-L90
- https://github.com/wechaty/wechaty-puppet-gitter/blob/96dcc41488bcce074284ed17496b03eb0175d285/src/puppet-gitter.ts#L540

Reply:
I test it with Stream and Buffer which doesn't work for me.  I will get into deeper test with your examples.